app/App.tsx

```diff
@@ -0,0 +1,79 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+
+import init from '../pkg/ezkl'
+
+import GenPK from './GenPk'
+import GenVk from './GenVk'
+import GenProof from './GenProof'
+import Verify from './Verify'
+import Hash from './Hash'
+
+interface Files {
+  [key: string]: File | null
+}
+
+export default function Home() {
+  const [files, setFiles] = useState<Files>({})
+
+  useEffect(() => {
+    async function run() {
+      // Initialize the WASM module
+      await init()
+    }
+    run()
+  })
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const id = event.target.id
+    const file = event.target.files?.item(0) || null
+    setFiles((prevFiles) => ({ ...prevFiles, [id]: file }))
+  }
+
+  return (
+    <div className='App'>
+      <GenPK
+        files={{
+          model: files['model_ser_pk'],
```

Comment:
you should be able to access this the normal way if you feel it's more readable `files.model_ser_pk`